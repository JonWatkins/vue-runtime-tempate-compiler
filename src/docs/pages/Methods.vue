<template>
  <div class="page">
    <h1 id="top" class="display-4 mb-3">
      <router-link to="/guide/methods#top" class="header-anchor">
        <span class="header-anchor--hash">#</span>
        Methods
      </router-link>
    </h1>

    <div class="alert alert-primary mb-4">
      These will behave as if they have been called from within the parent
      component.
    </div>

    <p>
      As with the <code>data</code> object the methods are also available in the
      <code>vue-runtime-template-compiler</code>. You just use them as if you
      are writing bindings in the parent's compiled template.
    </p>

    <pre class="language-html mb-4"><code>{{ example | trim }}</code></pre>

    <PageNav>
      <template v-slot:left>
        <router-link to="/guide/parent-context">
          <i class="fas fa-arrow-left"></i>
          Parent Context
        </router-link>
      </template>
      <template v-slot:right>
        <router-link to="/guide/computed">
          Computed
          <i class="fas fa-arrow-right"></i>
        </router-link>
      </template>
    </PageNav>
  </div>
</template>

<script>
import PageNav from '../components/PageNav.vue'

const example = `
<template>
  <runtime-template-compiler :template="template" \/>
<\/template>

<script>
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  data: () => ({
    userName: 'Bob',
    template: \`<div @click="logUser(userName)">Hello {{ userName }}!<\/div>\`,
  }),
  methods: {
    logUser(userName) {
      console.log(\`Users name is: \${userName}\`)
    }
  }
}
<\/script>
`

export default {
  name: 'Methods',

  components: {
    PageNav
  },

  data() {
    return {
      example
    }
  }
}
</script>
