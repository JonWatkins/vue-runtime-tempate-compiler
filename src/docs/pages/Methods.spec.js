import { shallowMount } from '@vue/test-utils'
import Methods from './Methods.vue'
const { localVueInst } = global

let wrapper

describe('Methods', () => {
  beforeEach(() => {
    wrapper = shallowMount(Methods, {
      localVue: localVueInst
    })
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('Methods')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
