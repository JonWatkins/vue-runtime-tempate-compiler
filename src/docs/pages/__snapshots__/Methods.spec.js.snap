// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Methods render correctly 1`] = `
<div class="page">
  <h1 id="top" class="mb-3">
    <router-link-stub to="/guide/methods#top" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      Methods
    </router-link-stub>
  </h1>
  <div class="alert alert-primary mb-4">
    These will behave as if they have been called from within the parent
    component.
  </div>
  <p>
    As with the <code>data</code> object the methods are also available in the
    <code>runtime-template-compiler</code>. You just use them as if you are
    writing bindings in the parent's compiled template.
  </p> <pre class="language-html mb-4"><code>&lt;template&gt;
  &lt;runtime-template-compiler :template="template" /&gt;
&lt;/template&gt;

&lt;script&gt;
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  data: () =&gt; ({
    userName: 'Bob',
    template: \`&lt;div @click="logUser(userName)"&gt;Hello {{ userName }}!&lt;/div&gt;\`,
  }),
  methods: {
    logUser(userName) {
      console.log(\`Users name is: \${userName}\`)
    }
  }
}
&lt;/script&gt;</code></pre>
  <pagenav-stub></pagenav-stub>
</div>
`;
